src/kernel/src/llvm/module.rs

```diff
@@ -0,0 +1,71 @@
+use super::Error;
+use llvm_sys::core::LLVMDisposeModule;
+use llvm_sys::execution_engine::{
+    LLVMCreateExecutionEngineForModule, LLVMDisposeExecutionEngine, LLVMExecutionEngineRef,
+};
+use llvm_sys::prelude::{LLVMContextRef, LLVMModuleRef};
+use std::ffi::c_char;
+use std::ptr::null_mut;
+use std::sync::{Mutex, MutexGuard};
+
+/// A wrapper on LLVM module for thread-safe.
+pub struct LlvmModule<'a> {
+    context: &'a Mutex<LLVMContextRef>,
+    raw: LLVMModuleRef,
+    _lock: MutexGuard<'a, LLVMContextRef>,
+}
```

Comment:
should have more consistency with LLVM name. Should be ```LLVMModule```.

Reply:
The problem is that one is not consistent with Rust naming. If you take a look on Rust `std` you will see they always use pascal-case on an abbreviation (e.g. `TypeId` instead of `TypeID`).